### PR TITLE
[Issue #2329] API-CI/CD Automation Conflict

### PR DESCRIPTION
**Pull request by babebe**
## Summary
Fixes [#{2329}](https://github.com/HHS/simpler-grants-gov/issues/2329)

## Changes proposed
Merge workflows into one file.

## Context for reviewers
Updates to PR  while running the jobs(ex. ERD diagram generated) would cancel the other job. Due to jobs being similar opted for running it under one action.

### CODE REVIEW

**Comment by chouinar** [comment]
@babebe - if you make a change to both the API folder and DB models in the same commit, does this work in parallel with your change? (And good find to look at the concurrency, I hadn't even thought of that when writing up the ticket)

**Comment by acouch** [comment]
Please make sure to follow the PR naming conventions before you wrap up: https://github.com/HHS/simpler-grants-gov/blob/main/DEVELOPMENT.md#title , thanks.

**Comment by babebe** [comment]
> Please make sure to follow the PR naming conventions before you wrap up: https://github.com/HHS/simpler-grants-gov/blob/main/DEVELOPMENT.md#title , thanks.
Yup, i will definitely do that.

**Comment by babebe** [comment]
> Please make sure to follow the PR naming conventions before you wrap up: https://github.com/HHS/simpler-grants-gov/blob/main/DEVELOPMENT.md#title , thanks.

Yup i will definitely do that

**Comment by babebe** [comment]
> @babebe - if you make a change to both the API folder and DB models in the same commit, does this work in parallel with your change? (And good find to look at the concurrency, I hadn't even thought of that when writing up the ticket)

@chouinar I just run a test with both changes and it run both.

**Comment by chouinar** [comment]
> > @babebe - if you make a change to both the API folder and DB models in the same commit, does this work in parallel with your change? (And good find to look at the concurrency, I hadn't even thought of that when writing up the ticket)
> 
> @chouinar I just run a test with both changes and it run both.

It looks like it failed when you did that because both tried to push to the branch: https://github.com/HHS/simpler-grants-gov/actions/runs/11407254033/job/31742847031

File: .github/workflows/ci-erd-openapi.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+# Update database ERD diagrams and OpenAPI docs so that they remain up to date with the application
+name: Update Database ERD Diagrams and OpenAPI Docs
 
 on:
   pull_request:
     paths:
       - api/**
       - Makefile
-      - .github/workflows/ci-openapi.yml
+      - .github/workflows/ci-erd-openapi.yml
```
Comment (coilysiren) [inline .github/workflows/ci-erd-openapi.yml:9]: Can you keep the filename as is 🙏🏽 it'll prevent merge conflicts if we end up wanting to pull in changes from the upstream that this file came from

Comment (babebe) [reply .github/workflows/ci-erd-openapi.yml]: makes sense. done

**Review by coilysiren — APPROVED** [review]
probably wait for @chouinar though

**Review by chouinar — APPROVED** [review]

**Pull request merged**